src/components/CategoriesGrid/index.tsx

```diff
@@ -1,10 +1,20 @@
 import React from 'react'
-import { useFeaturedCategories } from '../../utils/hooks/useFeaturedCategories'
+import { useSelector } from 'react-redux'
+import Category from '../../interfaces/Category'
+import { getCategories } from '../../store/categoriesSlice'
+import Spinner from '../Spinner'
 import CategoryItem from './CategoryItem'
 import './styles.scss'
 
 export default function CategoriesGrid () {
-  const { data, isLoading } = useFeaturedCategories()
+  const { data, isLoading } = useSelector(getCategories)
+  const { results }: { results: Array<Category> } = data
+
+  if (isLoading) {
```

Comment:
Be aware of 2 return in the same component, as it works fine, if component logic gets larger, would be more difficult to read, just a best practice